react-app-drawer-component/src/app-drawer.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+
+class AppDrawer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { isShown: false };
+    this.cssClass = 'side-menu';
+    this.handleClick = this.handleClick.bind(this);
+    this.hideMenu = this.hideMenu.bind(this);
+  }
+
+  handleClick(event) {
+    this.setState(function (state) {
+      if (!this.state.isShown) {
+        this.cssClass += ' show';
+        document.body.addEventListener('click', this.hideMenu);
+        document.body.style.backgroundColor = 'rgba(0, 0, 0, 0.4)';
+        return { isShown: true };
+      } else {
+        document.body.style.backgroundColor = 'white';
+        return { isShown: false };
+      }
+    });
+  }
+
+  hideMenu() {
+    this.setState(function () {
+      this.cssClass = 'side-menu';
+      document.body.style.backgroundColor = 'white';
+      return { isShown: false };
```

Comment:
As mentioned before, you don't want to reach outside of the component and affect other stuff on the page that was not created in the `render()` method.